lib/tarearbol/jobs/job.ex

```diff
@@ -133,6 +133,14 @@ defmodule Tarearbol.Job do
         delay(opts)
         retry_or_die(:on_error, job, opts, data, retries_left)
 
+      {_, {:ok, {:error, _, data, _}}} when is_struct(data) ->
```

Comment:
This breaks backward compatibility for Elixir ≤ 1.11, and this library is in use by the community.
```suggestion
      {_, {:ok, {:error, _, %_{} = data, _}}} ->
```